platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -116,30 +94,91 @@ class HangingProtocolService {
 
   public reset() {
     this.studies = [];
-    this.protocols = [];
+    this.protocols = new Map();
     this.hpAlreadyApplied = [];
     this.viewportMatchDetails = [];
     // this.ProtocolEngine.reset()
   }
 
-  public getMatchDetails(): HangingProtocolMatchDetails {
+  public getDefaultProtocol(): HangingProtocol.Protocol {
+    return this.getProtocolById('default');
+  }
+
+  public getMatchDetails(): HangingProtocol.HangingProtocolMatchDetails {
     return {
       viewportMatchDetails: this.viewportMatchDetails,
       displaySetMatchDetails: this.displaySetMatchDetails,
       hpAlreadyApplied: this.hpAlreadyApplied,
     };
   }
 
-  public getProtocols() {
-    return this.protocols;
+  /**
+   * It loops over the protocols map object, and checks whether the protocol
+   * is a function, if so, it executes it and returns the result as a protocol object
+   * otherwise it returns the protocol object itself
+   *
+   * @returns all the hanging protocol registered in the HangingProtocolService
+   */
+  public getProtocols(): HangingProtocol.Protocol[] {
+    // this.protocols is a map of protocols with the protocol id as the key
+    // and the protocol or a function that returns a protocol as the value
+    const protocols = [];
+    // @ts-ignore
+    for (const protocolId of this.protocols.keys()) {
+      const protocol = this.getProtocolById(protocolId);
+      if (protocol) {
+        protocols.push(protocol);
+      }
+    }
+
+    return protocols;
   }
 
-  public addProtocols(protocols) {
-    protocols.forEach(protocol => {
-      if (this.protocols.indexOf(protocol) === -1) {
-        this.protocols.push(this._validateProtocol(protocol));
+  /**
+   * Returns the protocol with the given id
```

Comment:
Please add documentation that it will call the generator function to create/update the protocol as necessary.